components/nexus-security/src/main/java/org/apache/shiro/nexus/ShiroJCacheAdapter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.apache.shiro.nexus;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.cache.Cache.Entry;
+
+import org.sonatype.sisu.goodies.common.ComponentSupport;
+
+import org.apache.shiro.cache.Cache;
+import org.apache.shiro.cache.CacheException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Shiro {@link javax.cache.Cache} to {@link Cache} adapter.
+ *
+ * @since 3.0
+ */
+public class ShiroJCacheAdapter<K, V>
+  extends ComponentSupport
+  implements Cache<K, V>
+{
+  private final javax.cache.Cache<K, V> cache;
+
+  private final String name;
+
+  public ShiroJCacheAdapter(final javax.cache.Cache<K, V> cache) {
+    this.cache = checkNotNull(cache);
+    this.name = cache.getName();
+  }
+
+  @Override
+  public V get(final K key) throws CacheException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the declaration of thrown exception 'org.apache.shiro.cache.CacheException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)